### DISCUSSION TITLE
New Source: Postmark

### DISCUSSION THREAD

**Discussion by benweissbehavehealth** [airbytehq/airbyte#33145] [New Connector Request]
## Tell us about the new connector you’d like to have
* Which source and which destination?

**Postmark - https://postmarkapp.com/developer/api/overview**

## Describe the context around this new connector
* Why do you need this integration? How does your team intend to use the data? This helps us understand the use case.
We want to have a copy of all the data created in Postmark for the purpose of backup and archiving.  This would be very similar to any other email service connector like Sendgrid.

* If this is an API source connector, which entities/endpoints do you need supported?

Servers
https://postmarkapp.com/developer/api/server-api#get-server
https://postmarkapp.com/developer/api/servers-api#get-server

Messages Streams
https://postmarkapp.com/developer/api/message-streams-api#list-message-streams
https://postmarkapp.com/developer/api/message-streams-api#get-a-message-stream

Outbound Messages
https://postmarkapp.com/developer/api/messages-api#outbound-message-dump

Inbound Messages
https://postmarkapp.com/developer/api/messages-api#inbound-message-search
https://postmarkapp.com/developer/api/messages-api#inbound-message-details

Message Activity
https://postmarkapp.com/developer/api/messages-api#message-opens
https://postmarkapp.com/developer/api/messages-api#message-clicks

Webhooks
https://postmarkapp.com/developer/api/webhooks-api#list-webhooks
https://postmarkapp.com/developer/api/webhooks-api#get-a-webhook

**Comment by sherifnada** [comment]
thanks @benweissbehavehealth for the request! Please follow on this issue for any updates

**Comment by karinakuz** [comment]
**Postmark Integration Vetting**


Webhook-based? (no/partially/yes) if partially, mention which endpoints are webhook-based 
_Webhook and REST API are available_ 
_https://postmarkapp.com/developer/webhooks/webhooks-overview_

Available authentication modes (API key/Oauth/other) 
_Postmark has two types of API tokens: Server Token and Account Token_ _https://postmarkapp.com/developer/api/overview#authentication_

Connecting to the data source what kind of API is it? (SOAP/GraphQL/REST/other)
 _REST API https://postmarkapp.com/developer/api/overview_

is there an SDK available? Can we connect with HTTP requests directly?
 _no_

If this is a database/datalake/pubsub source, does it have a JDBC driver or Java SDK? 
_no_

Creating an account Select one of the below options self-service free account available?
_Free sandbox_

How to populate the account with data?
* Can it be done entirely through the API? 
_yes_
* Will it require some work through the UI? If so, what is required? 
_no_
* Will it require performing some paid action like running an Ad campaign? If so, please note the action needed, estimated cost, and estimated effort in hours. If a task will take more than 4 hours, please break it down into smaller tasks each of which take no more than 4 hours 
_no_
* Do you need any actions from Airbyte to populate the account with data? If so, please tag @ sherifnada for support 
_no_

Available streams for sync
* If this is an API connector:
Provide a link  to the reference docs 
_https://postmarkapp.com/developer/api/overview_
Servers https://postmarkapp.com/developer/api/servers-api
Messages https://postmarkapp.com/developer/api/messages-api
Message Streams https://postmarkapp.com/developer/api/message-streams-api
Webhooks https://postmarkapp.com/developer/api/webhooks-api

Please mention if any of the endpoints are asynchronous (meaning we have to make a request, wait for the server to compute a response, then make another request to obtain the result)
_no_

If this is a source where the schema is dynamic/not fixed e.g: database, mention the data model of the connector i.e: relational, document-based, graph-based, etc 
_no_

Do any streams support incremental sync?
not necessary to mention all of them, just saying “none”, “some”, or “all” is sufficient. If only some streams, please mention which ones support incremental sync 
_Messages Inound/Otound, Message Opens/Clicks, Message Streams_

**Comment by karinakuz** [comment]
**Sandbox with data is created, credentials are saved in the LastPass with the name with name "Postmark", tested APIs with tokens for the main streams:**_
- Servers https://api.postmarkapp.com/server
- Server by ID https://api.postmarkapp.com/servers/8429436
- Message Streams https://api.postmarkapp.com/message-streams?MessageStreamType=all&IncludeArchivedStreams=true
- Message Stream by ID https://api.postmarkapp.com/message-streams/airbyte-messages
- Outbound Message Dump https://api.postmarkapp.com/messages/outbound/ac81c8b7-29f8-4442-9b52-b1c6241c9aa6/dump
- Outbound Message Details https://api.postmarkapp.com/messages/outbound/458b7c3f-34d1-4ab7-9b54-359de033c649/details
- Message Opens https://api.postmarkapp.com/messages/outbound/opens?count=50&offset=0
- Webhooks https://api.postmarkapp.com/webhooks?MessageStream=outbound
- Webhook by ID https://api.postmarkapp.com/webhooks/10419742

**Schemas:** https://drive.google.com/drive/folders/1WYcGO1jU8C_ALQD8xO7Rx6ID79WgI7_8?usp=sharing

[a comment by an automated account is not shown]
